### PR TITLE
fix(components): [upload] photo-wall wrap

### PR DESCRIPTION
**Pull request by YunYouJun**
- use `h` rewrite uploadContentWithProps
- fix `upload` photo-wall flex wrap

before:
image
image

after:
image

---

Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/upload/src/upload.vue [hunk 1]
```diff
@@ -114,4 +100,49 @@ defineExpose({
...
+        onExceed: props.onExceed,
+        onStart: handleStart,
+        onProgress: handleProgress,
+        onSuccess: handleSuccess,
+        onError: handleError,
+        onRemove: handleRemove,
+      },
+      {
+        default: () => {
+          if (slots.trigger) {
```
Comment (HerringtonDarkholme) [inline packages/components/upload/src/upload.vue:138]: Is this equivalent to something like:

```suggestion
<UploadContent v-bind="props">
  <slot name="trigger" v-if="slots.trigger"/>
  <slot v-if="slots.default"/>
</UploadContent>
```

Comment (YunYouJun) [reply packages/components/upload/src/upload.vue:138]: But not all props were passed in `UploadContent`.

If use template, I have to maintain it in two places.

Comment (HerringtonDarkholme) [reply packages/components/upload/src/upload.vue]: But creating a component on the fly has undesirable performance impact. For UI framework,  we should prefer end users' performance over slight duplication. Especially the burden isn't large.

Comment (YunYouJun) [reply packages/components/upload/src/upload.vue]: Ok. I rewrite it with `template`.

File: packages/components/upload/src/upload.vue [hunk 2]
```diff
@@ -114,4 +115,34 @@ defineExpose({
...
+  showFileList: props.showFileList,
+  httpRequest: props.httpRequest,
+  beforeUpload: props.beforeUpload,
+  onExceed: props.onExceed,
+  onStart: handleStart,
+  onProgress: handleProgress,
+  onSuccess: handleSuccess,
+  onError: handleError,
+  onRemove: handleRemove,
+}
```
Comment (JeremyWuuuuu) [inline packages/components/upload/src/upload.vue:147]:
```suggestion
const uploadContentProps: UploadContentProps = computed(() => ({
  type: props.type,
  drag: props.drag,
  action: props.action,
  multiple: props.multiple,
  withCredentials: props.withCredentials,
  headers: props.headers,
  method: props.method,
  name: props.name,
  data: props.data,
  accept: props.accept,
  autoUpload: props.autoUpload,
  listType: props.listType,
  disabled: props.disabled,
  limit: props.limit,
  fileList: props.fileList as UploadFiles,
  showFileList: props.showFileList,
  httpRequest: props.httpRequest,
  beforeUpload: props.beforeUpload,
  onExceed: props.onExceed,
  onStart: handleStart,
  onProgress: handleProgress,
  onSuccess: handleSuccess,
  onError: handleError,
  onRemove: handleRemove,
}))
```

**Review by jw-foss — APPROVED** [review]

**Pull request merged**